src/NuGet.Core/NuGet.Repositories/NuGetv3LocalRepository.cs

```diff
@@ -76,6 +76,21 @@ public IEnumerable<LocalPackageInfo> FindPackagesById(string packageId)
                 });
         }
 
+        /// <summary>
+        /// Remove cached results for the given ids. This is needed
+        /// after installing a new package.
+        /// </summary>
+        public void ClearCacheForIds(IEnumerable<string> packageIds)
+        {
+            lock (_cache)
```

Comment:
Is there a reason why we use a lock on a dictionary and not use a ConcurrentDictionary? I understand that it is a question on the existing code. Just curious